charts/perses/templates/statefulset.yaml

```diff
@@ -197,4 +197,36 @@ spec:
             defaultMode: 420
         {{- end }}
         {{- end }}
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- if or .Values.podAntiAffinity .Values.podAntiAffinityTopologyKey }}
+      affinity:
+        podAntiAffinity:
+          {{- if .Values.podAntiAffinity }}
+          {{- toYaml .Values.podAntiAffinity | nindent 10 }}
+          {{- else }}
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+                matchExpressions:
+                  - key: app.kubernetes.io/name
+                    operator: In
+                    values:
+                      - {{ include "perses.name" . }}
+              topologyKey: {{ .Values.podAntiAffinityTopologyKey | quote }}
+          {{- end }}
+      {{- end }}
```

Comment:
This condition always evaluates to `true` because `podAntiAffinityTopologyKey` has a default value (`"kubernetes.io/hostname"`) in values.yaml, even when `podAntiAffinity` is empty (`""`). This makes the podAffinity block non-optional. This logic should be opt-in.